azurerm/internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -1098,6 +1102,15 @@ func flattenKubernetesClusterDataSourceManagedClusterIdentity(input *containerse
 		identity["tenant_id"] = *input.TenantID
 	}
 
+	identity["user_assigned_identity_id"] = ""
+	if input.UserAssignedIdentities != nil {
+		keys := []string{}
+		for key := range input.UserAssignedIdentities {
+			keys = append(keys, key)
```

Comment:
Can you run the identity through the parser here to avoid possible casing issues? ref #10105, #10031 & #10162

Reply:
done in https://github.com/flo-02-mu/terraform-provider-azurerm/pull/2, 

specific commit is https://github.com/flo-02-mu/terraform-provider-azurerm/commit/8e698d40effc4810d51cb410d77af21f7154d50c

The branch is quite updated so I had to merge with master to use the ID parsing